src/common/helpers/requestMetaData/getRequestMetaData.ts

```diff
@@ -0,0 +1,79 @@
+import { UMIP } from "features/vote/helpers/fetchUMIP";
+import {
+  decodeAncillaryDataHexString,
+  getTitleFromAncillaryData,
+  getDescriptionFromAncillaryData,
+} from "./ancillaryData";
+import approvedIdentifiers from "./approvedIdentifiersTable.json";
+
+type RequestMetaData = {
+  title: string;
+  description: string;
+  umipUrl?: string;
+};
+
+/** Finds a title and description, and UMIP link (if it exists) for a proposal.
+ *
+ * There are 3 different sources of this data, depending on the proposal type:
+ *
+ * 1. For UMIPs, the title, description and UMIP link come from Contentful.
+ * 2. For requests for approved price identifiers, the title, description, and UMIP link come from the hard-coded `approvedIdentifiersTable` json file.
+ * 3. For requests from Polymarket, the title and description come from the proposal's ancillary data (note that there is no UMIP link here).
+ */
+export function getRequestMetaData(
+  ancillaryData: string,
+  proposal: string,
+  umip?: UMIP
+): RequestMetaData {
+  // if we are dealing with a UMIP, get the title, description and UMIP url from Contentful
+  const isUmip = proposal.includes("Admin");
+  if (isUmip) {
+    return {
+      title: umip?.title ?? proposal,
+      description:
+        umip?.description ?? "No description was found for this UMIP.",
+      umipUrl: umip?.umipLink,
+    };
+  }
+
+  // if we are dealing with a request for an approved price identifier, get the title, description and UMIP url from the hard-coded approvedIdentifiersTable json file
+  // we know we are dealing with a request for an approved price identifier if the proposal matches an approved identifier's title
+  const identifierDetails = approvedIdentifiers.find(
+    (id) => proposal === id.title
+  );
+  const identifiersToIgnore = [
```

Comment:
This is totally a minor nit, but ignore seems like an odd term. it's almost like the opposite: we're selecting it to extract info from the ancillary data. Ignoring sounds like we're going to exclude it from the UI or something. I have no problems with the actual code here, it just conceptually feels a little off. No changes necessarily required if you're happy with this language, just wanted to note that it was a bit confusing to me as a reader.

Reply:
Thanks very much for the explanation! I was trying to ask if there are any other special cases, i.e. approved identifiers that Polymarket uses which we should rather treat as Polymarket requests. 

Since `YES_OR_NO_QUERY` is the only one, and I do agree that the language I used was confusing, I've changed the implementation to simply treat `YES_OR_NO_QUERY` as a Polymarket request instead. I've removed the confusingly named array, and added a comment explaining the situation.